### PR TITLE
Complete implementation

### PR DESCRIPTION
**Pull request by timanglade**
The convolutional layers were missing. I also added the batch normalization layers.

### CODE REVIEW

**Comment by timanglade** [comment]
I double-checked the layer input shapes against the paper at alpha=1, and it seems correct. The model also seems to train/converge correctly on a dataset I tried it on.

**Comment by rcmalli** [comment]
Thanks! I wasn't sure about standard convolution layers just before depthwise convolution ones so, I didn't add them. According to Figure 3 on the paper, there are also batchnorm layers between depthwise and pointwise layers which we use them as a single layer called "SeparableConv", but I think they should be added from TF backend.

**Comment by timanglade** [comment]
@rcmalli Oh I see, I may be doing this wrong then. I was under the impression `SeparableConv2D` was just doing the depthwise convolution, but after re-reading the docs it does seem to do both.

**Comment by rcmalli** [comment]
I am reverting this merge for now. According to Tensorflow documentation, there is already depthwise convolution implementation. Implementing it with tf backend as a custom Keras layer should able us to add batchnorm layers after each of them. [depthwise_conv2d](https://www.tensorflow.org/api_docs/python/tf/nn/depthwise_conv2d)

**Comment by timanglade** [comment]
Yup, that sounds right, sorry about this! The BN+ReLU after the SeparableConv2D should be correct though, right? Just missing a way to add BN+ReLU between dw and pw convs.

**Comment by rcmalli** [comment]
Yes, I think that's true. I have updated the code and added BN+ReLU after each SeparableConv2D and the first Convolution2D layer.

**Comment by rcmalli** [comment]
@timanglade I tried to implement the standalone depthwise convolution layer and updated the source code according to it but still,l it should be tested with training.

**Comment by timanglade** [comment]
@rcmalli Cool, I will try to give it a spin in a bit, but for now I’m just using your previous implementation (without the BN+ReLU between dw and pw convolutions) based on a [tip from the author of the QuickNet paper](https://www.reddit.com/r/MachineLearning/comments/663m43/r_170404861_mobilenets_efficient_convolutional/dgfaoz1/?utm_content=permalink&utm_medium=front&utm_source=reddit&utm_name=MachineLearning). Once that training is completed, I may give your implementation a try and see if accuracy is indeed lower that way.
